src/towncrier/test/helpers.py

```diff
@@ -69,3 +70,43 @@ def setup_simple_project(
 
     if mkdir_newsfragments:
         Path("foo/newsfragments").mkdir()
+
+
+def with_isolated_simple_project(
+    *,
+    with_git=False,
```

Comment:
I am not a big fan of flag arguments.

I prefer something like `@with_project` and `@with_git_project` . but we can have this like this, as long as the intended usage of `with_git` is documented.